client/lib/site/index.js

```diff
@@ -100,6 +100,16 @@ Site.prototype.replace = function( attributes ) {
 
 	let changed = false;
 
+	// Set attrs which will get computed and will mean the same before doing equality checks
+	// so the 'change' event is not triggered. See Site.prototype.updateComputedAttributes.
+	if ( attributes.hasOwnProperty( 'options' ) ) {
+		let options = attributes.options;
+
+		if ( ! options.hasOwnProperty( 'default_post_format' ) || options.default_post_format === '0' ) {
+			options.default_post_format = 'standard';
+		}
+	}
+
```

Comment:
So if you look at `updateComputedAttributes` anything falsy for `default_post_format` is equivalent to 'standard'. Which means that undefined `default_post_format` should get set to 'standard' too.

```javascript
const computedAttrsCustomizer = ( newAttrVal, siteAttrVal, attrKey ) => {
    // options.default_post_format
    if ( attrKey === 'default_post_format' ) {
        const normalizedNewAttrVal = ! newAttrVal ? ' standard' : newAttrVal;
        const normalizedsiteAttrVal = ! siteAttrVal ? ' standard' : siteAttrVal;
        return normalizedNewAttrVal === normalizedsiteAttrVal;
    }
    // Use the default comparison mechanism.
    return;
};
// if we compare a and b
a.options.default_post_format = a.options.default_post_format || false;
b.options.default_post_format = b.options.default_post_format || false;
isEqualWith( a, b, computedAttrsCustomizer );
```